mir_eval/io.py

```diff
@@ -477,3 +477,86 @@ def load_key(filename, delimiter=r'\s+'):
         warnings.warn(error.args[0])
 
     return key_string
+
+
+def load_ragged_time_series(filename, dtype=float, delimiter=r'\s+',
+                            header=False):
+    r"""Utility function for loading in data from a delimited time series
+    annotation file with a variable number of columns.
+    Assumes that column 0 contains time stamps and columns 1 through n contain
+    values. n may be variable from time stamp to time stamp.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the annotation file
+    dtype : function
+        Data type to apply to values columns.
+    delimiter : str
+        Separator regular expression.
+        By default, lines will be split by any amount of whitespace.
+    header : bool
+        Indicates whether a header row is present or not.
+        By default, assumes no header is present.
+
+    Returns
+    -------
+    times : np.ndarray
+        array of timestamps (float)
+    values : list of np.ndarray
+        list of arrays of corresponding values
+
+    """
+    # Initialize empty lists
+    times = []
+    values = []
+
+    # Create re object for splitting lines
+    splitter = re.compile(delimiter)
+
+    # Keep track of whether we create our own file handle
+    own_fh = False
+    # If the filename input is a string, need to open it
+    if isinstance(filename, six.string_types):
+        # Remember that we need to close it later
+        own_fh = True
+        # Open the file for reading
+        input_file = open(filename, 'r')
+    # If the provided has a read attribute, we can use it as a file handle
+    elif hasattr(filename, 'read'):
+        input_file = filename
+    # Raise error otherwise
+    else:
+        raise ValueError('filename must be a string or file handle')
+    if header:
+        start_row = 1
+    else:
+        start_row = 0
+    for row, line in enumerate(input_file, start_row):
+        # Split each line using the supplied delimiter
+        data = splitter.split(line.strip())
+        try:
+            converted_time = float(data[0])
+        except:
```

Comment:
avoid catch-all exceptions; only catch the exception you expect here, let everything else float up-stream.